gateway/session_manager.go

```diff
@@ -234,41 +237,61 @@ func (l *SessionLimiter) ForwardMessage(r *http.Request, session *user.SessionSt
 		return sessionFailRateLimit
 	}
 
+	var (
+		apiLimit            = accessDef.Limit.Clone()
+		endpointRLKeySuffix = ""
+	)
+
+	if len(accessDef.Endpoints) > 0 {
+		endpointRLInfo, doEndpointRL := getEndpointRateLimitInfo(r.Method, r.URL.Path, accessDef.Endpoints)
+		if doEndpointRL {
+			apiLimit.Rate = endpointRLInfo.rate
```

Comment:
Create a func APILimit() on the type, lets keep the logic of converting the type in one place (low number of symbols, small func to extract this)